### PR TITLE
fix: correctly escape quotes in field names

### LINKED ISSUE #595: Quotes in field names cause preprocessing errors
Field names are not properly quoted when generating the queries for DuckDB. They're surrounded by double quotes without escaping (by doubling) any double quotes already in the name.

https://github.com/GenSpectrum/LAPIS-SILO/blob/fb58525de5d978075defbe4a22d5a3f37dbb6d44/src/silo/preprocessing/metadata_info.cpp#L160-L191

It looks like the code above should be using `Identifier::escapeIdentifier()` instead, although I suspect doing so will require other downstream changes in calling code.

I noticed this when reading the preprocessing code and then verified it with the [attached test input directory](https://github.com/user-attachments/files/17119200/tsvWithQuoteInFieldName.tar.gz) and the following command:

```console
$ docker run --rm -v "$PWD"/testBaseData/tsvWithQuoteInFieldName:/preprocessing/input:ro -v "$PWD"/testBaseData/output:/preprocessing/output:rw -v "$PWD"/testBaseData/tsvWithQuoteInFieldName/preprocessing_config.yaml:/app/preprocessing_config.yaml:ro -v "$PWD"/testBaseData/tsvWithQuoteInFieldName/database_config.yaml:/app/database_config.yaml:ro -v "$PWD"/testBaseData/logs:/app/logs:rw ghcr.io/genspectrum/lapis-silo:0.2.21 --preprocessing
[2024-09-24 18:13:24.355] [logger] [info] [api.cpp:316] Starting SILO
[2024-09-24 18:13:24.356] [logger] [info] [api.cpp:274] Starting SILO preprocessing
[2024-09-24 18:13:24.356] [logger] [info] [yaml_file.cpp:11] Reading config from ./default_preprocessing_config.yaml
[2024-09-24 18:13:24.356] [logger] [info] [yaml_file.cpp:11] Reading config from ./preprocessing_config.yaml
[2024-09-24 18:13:24.356] [logger] [info] [api.cpp:72] Resulting preprocessing config: { input directory: '/preprocessing/input/', pango_lineage_definition_file: none, output_directory: '/preprocessing/output/', metadata_file: ''metadata.tsv'', reference_genome_file: 'reference_genomes.json',  gene_file_prefix: 'gene_',  nucleotide_sequence_file_prefix: 'nuc_', unaligned_nucleotide_sequence_file_prefix: 'unaligned_', ndjson_filename: none, preprocessing_database_location: none }
[2024-09-24 18:13:24.356] [logger] [info] [database_config.cpp:211] Reading database config from database_config.yaml
[2024-09-24 18:13:24.356] [logger] [info] [api.cpp:257] preprocessing - reading reference genome
[2024-09-24 18:13:24.356] [logger] [info] [reference_genomes.cpp:159] Read reference genomes from file: /preprocessing/input/reference_genomes.json
[2024-09-24 18:13:24.356] [logger] [info] [api.cpp:261] preprocessing - reading pango lineage alias
[2024-09-24 18:13:24.356] [logger] [info] [pango_lineage_alias.cpp:106] No pango lineage alias file provided. Using empty alias lookup.
[2024-09-24 18:13:24.363] [logger] [info] [preprocessor.cpp:85] preprocessing - creating intermediate results directory '/preprocessing/temp/'
[2024-09-24 18:13:24.363] [logger] [info] [preprocessor.cpp:115] preprocessing - classic metadata file pipeline chosen
[2024-09-24 18:13:24.364] [logger] [error] [api.cpp:282] Parser Error: unterminated quoted identifier at or near "" VARCHAR);"
LINE 1: ... TABLE metadata_table("id" VARCHAR,"x"y" VARCHAR);
                                                  ^
[2024-09-24 18:13:24.364] [logger] [info] [api.cpp:321] Stopping SILO
std::exception
```

### PR DESCRIPTION
**Pull request by Taepper**
resolves #595 

### Summary

We did not correctly escape quotes in all cases. The detected error is added as a test and the corresponding bug is now fixed.

## PR Checklist

~~- [ ] All necessary documentation has been adapted or there is an issue to do so.~~
- [X] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Review by fengelniederhammer — COMMENTED** [review]
I'm getting an error:

```
[2024-09-26 13:57:41.395] [logger] [debug] [preprocessing_database.cpp:73] Preprocessing Database - Result:
Parser Error: syntax error at or near ""lineage IS NULL THEN 'NULL'::VARCHAR ELSE '_' || pango_""
LINE 4: FROM (SELECT CASE WHEN pango_"lineage IS NULL THEN 'NULL'::VARCHAR ELSE '_' || pango_"lineage END AS partition_key, COUNT(*) AS count
      FROM metadata_table
      GROUP BY partition_key
      ORDER BY partition_key);
...
                                     ^

[2024-09-26 13:57:41.400] [logger] [error] [api.cpp:282] Parser Error: syntax error at or near ""lineage IS NULL THEN 'NULL'::VARCHAR ELSE '_' || pango_""
LINE 4: FROM (SELECT CASE WHEN pango_"lineage IS NULL THEN 'NULL'::VARCHAR ELSE '_' || pango_"lineage END AS partition_key, COUNT(*) AS count
      FROM metadata_table
      GROUP BY partition_key
      ORDER BY partition_key);
...
                                     ^
```

With `testBaseData/exampleDataset/database_config.yaml` changed to
```
schema:
  instanceName: sars_cov-2_minimal_test_config
  metadata:
    - name: gisaid_epi_isl
      type: string
    - name: date
      type: date
    - name: unsorted_date
      type: date
    - name: region
      type: string
      generateIndex: true
    - name: country
      type: string
      generateIndex: true
    - name: 'pango_"lineage'
      type: pango_lineage
    - name: division
      type: string
      generateIndex: true
    - name: age
      type: int
    - name: qc_value
      type: float
    - name: test_boolean_column
      type: boolean
  primaryKey: gisaid_epi_isl
  dateToSortBy: date
  partitionBy: 'pango_"lineage'
defaultNucleotideSequence: "main"
```
and the respective `small_metadata_set.tsv` header
`gisaid_epi_isl	pango_"lineage	date	region	country	division	unsorted_date	age	qc_value	test_boolean_column
`

File: src/silo/preprocessing/metadata_info.test.cpp [hunk 1]
```diff
@@ -53,11 +53,19 @@ TEST(MetadataInfo, isValidMedataFileShouldReturnTrueWithValidMetadataFile) {
          }
    };
 
-   const auto fields = silo::preprocessing::MetadataInfo::getMetadataFields(valid_config);
-   ASSERT_TRUE(std::ranges::find(fields, R"("gisaid_epi_isl")") != fields.end());
-   ASSERT_TRUE(std::ranges::find(fields, R"("pango_lineage")") != fields.end());
-   ASSERT_TRUE(std::ranges::find(fields, R"("date")") != fields.end());
-   ASSERT_TRUE(std::ranges::find(fields, R"("country")") != fields.end());
+   const auto raw_fields =
```
[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/preprocessing/metadata_info.test.cpp:56]: Can you elaborate?

[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/preprocessing/metadata_info.test.cpp:56]: Should I leave these extra checks out? In theory this file is not testing the escaping part, but simply that the validation is correct and the contained fields are correct. Should I only check the escaped fields then? Or only the not-escaped ones? We do not have equality for `class Identifier`, should I implement that instead?

[the remark this answers is not in the document]
Comment (fengelniederhammer) [reply src/silo/preprocessing/metadata_info.test.cpp:56]: Hm. Actually my comment was pretty unrelated to your code changes.

But I think it makes sense to test it, if it's not too hard. Maybe you can test for equality of the escaped values? (Then it's strings - they can be compared).

**Review by pflanze — COMMENTED** [review]
I don't see anything wrong with your changes--assuming that explicit escaping is the proper pragmatic decision to be taken right now. The safer solution is to never create SQL strings but use prepared statements and let the database handle correct value inclusion. Apparently DuckDB supports them[1]. This will be a somewhat larger change, depending on whether you want to keep the prepared statements around (may or may not be relevant for performance). Considering our tight time budget right now I guess it's fair to postpone that change.

[1] https://duckdb.org/docs/sql/query_syntax/prepared_statements.html

**Review by pflanze — APPROVED** [review]

**Pull request merged**